rotkehlchen/inquirer.py

```diff
@@ -388,8 +434,93 @@ def find_usd_price(
+            assert token is not None
+            result = get_underlying_asset_price(token)
+            usd_price = Price(ZERO) if result is None else Price(result)
+            Inquirer._cached_current_price[cache_key] = CachedPriceEntry(
+                price=usd_price,
+                time=ts_now(),
+            )
+            return usd_price
+
         return instance._query_oracle_instances(from_asset=asset, to_asset=A_USD)
 
+    def find_uniswap_v2_lp_price(
+        self,
+        token: Union[EthereumToken, UnknownEthereumToken],
+    ) -> Optional[Price]:
+        """
+        Calculate price for a LP token,this is
+        value = (Total value of liquidity pool)/(Current suply of LP tokens)
+        We need:
+        - Price of token 0
+        - Price of token 1
+        - Pooled amount of token 0
+        - Pooled amount of token 1
+        - Total supply of of pool token
+        """
+        assert self._ethereum is not None, 'Inquierer is not correctly initializen'
+
+        address = token.ethereum_address
+        contract = EthereumContract(address=address, abi=UNISWAP_V2_LP_ABI, deployed_block=0)
+        methods = ['token0', 'token1', 'totalSupply', 'getReserves', 'decimals']
+        try:
+            output = multicall_2(
+                ethereum=self._ethereum,
+                require_success=True,
+                calls=[(address, contract.encode(method_name=method)) for method in methods],
+            )
+        except RemoteError:
+            log.error(
+                f'Remote error calling multicall contract for uniswap v2 lp '
+                f'token {token.ethereum_address} properties',
+            )
+            return None
+
+        # decode output
+        decoded = []
+        for (method_output, method_name) in zip(output, methods):
+            if method_output[0] and len(method_output[1]) != 0:
+                decoded_method = contract.decode(method_output[1], method_name)
+                if len(decoded_method) == 1:
+                    decoded.append(decoded_method[0])
+                else:
+                    decoded.append(decoded_method)
+            else:
+                log.debug(
+                    f'Multicall to Uniswap V2 LP failed to fetch field {method_name} '
+                    f'for token {token.ethereum_address}',
+                )
+                return None
+
+        try:
+            token0 = EthereumToken(decoded[0])
+            token1 = EthereumToken(decoded[1])
+        except UnknownAsset:
+            return None
+
+        try:
+            token0_supply = FVal(decoded[3][0] * 10**-token0.decimals)
+            token1_supply = FVal(decoded[3][1] * 10**-token1.decimals)
+            total_supply = FVal(decoded[2] * 10 ** - decoded[4])
+        except ValueError as e:
+            log.debug(
+                f'Failed to convert token amounts into FVal for token {address} '
+                f'with values {str(decoded)}. f{str(e)}',
+            )
```

Comment:
This is an error and you should get out of the function. Otherwise you end up with undefined variables.

```suggestion
            log.error(
                f'Failed to deserialize token amounts for token {address} '
                f'with values {str(decoded)}. f{str(e)}',
            )
            return None
```

Please notice the addition of the return statement.